src/main/java/delight/nashornsandbox/internal/NashornSandboxImpl.java

```diff
@@ -39,9 +35,40 @@
 @SuppressWarnings("restriction")
 public class NashornSandboxImpl implements NashornSandbox {
 
+	private static final Class<?> JDK_NASHORN_NashornScriptEngineFactory_CLASS;
+	private static final Class<?> JDK_NASHORN_ClassFilter_CLASS;
+	private static final Class<?> STANDALONE_NASHORN_NashornScriptEngineFactory_CLASS;
+	private static final Class<?> STANDALONE_NASHORN_ClassFilter_CLASS;
+
+	static {
+		Class<?> tmp_JDK_NASHORN_NashornScriptEngineFactory_CLASS = null;
+		Class<?> tmp_JDK_NASHORN_ClassFilter_CLASS = null;
+		// TODO what behavior do we want here?
+		// TODO move all JDK/standalone code to some dedicated class?
+		try {
+			tmp_JDK_NASHORN_NashornScriptEngineFactory_CLASS = Class.forName("jdk.nashorn.api.scripting.NashornScriptEngineFactory");
+			tmp_JDK_NASHORN_ClassFilter_CLASS = Class.forName("jdk.nashorn.api.scripting.ClassFilter");
+		} catch (ClassNotFoundException e) {
+			System.out.println("JDK Nashorn not found");
+		}
+		JDK_NASHORN_NashornScriptEngineFactory_CLASS = tmp_JDK_NASHORN_NashornScriptEngineFactory_CLASS;
+		JDK_NASHORN_ClassFilter_CLASS = tmp_JDK_NASHORN_ClassFilter_CLASS;
+
+		Class<?> tmp_STANDALONE_NASHORN_NashornScriptEngineFactory_CLASS = null;
+		Class<?> tmp_STANDALONE_NASHORN_ClassFilter_CLASS = null;
+		try {
+			tmp_STANDALONE_NASHORN_NashornScriptEngineFactory_CLASS = Class.forName("org.openjdk.nashorn.api.scripting.NashornScriptEngineFactory");
+			tmp_STANDALONE_NASHORN_ClassFilter_CLASS = Class.forName("org.openjdk.nashorn.api.scripting.ClassFilter");
+		} catch (ClassNotFoundException e) {
+			System.out.println("Standalone Nashorn not found");
+		}
+		STANDALONE_NASHORN_NashornScriptEngineFactory_CLASS = tmp_STANDALONE_NASHORN_NashornScriptEngineFactory_CLASS;
+		STANDALONE_NASHORN_ClassFilter_CLASS = tmp_STANDALONE_NASHORN_ClassFilter_CLASS;
+	}
+
```

Comment:
Probably better to manage this in a centralised place?

Reply:
Most certainly, that should move to a class like NashornDetection, that would do the detection and debug logs about what is going on.